File: scratch/extract_clean_brochures.py
import re
from bs4 import BeautifulSoup

html_path = r'd:/xampp/htdocs/sssu/satya-sai/assets/images/sssutms.co.in/cms/Website/Admission/Brochures.html'
with open(html_path, 'r', encoding='utf-8', errors='ignore') as f:
    html = f.read()

soup = BeautifulSoup(html, 'html.parser')

print("=== PAGE TITLE ===")
title = soup.find('title')
if title:
    print(title.get_text(strip=True))

print("\n=== MAIN CARD HEADINGS AND PARAGRAPHS ===")
card_body = soup.find('div', class_='card-body')
if card_body:
    for elem in card_body.find_all(['h1', 'h2', 'h3', 'h4', 'h5', 'h6', 'p', 'a', 'table']):
        if elem.name == 'a':
            href = elem.get('href', '')
            text = elem.get_text(strip=True)
            print(f"LINK: Text='{text}' | Href='{href}'")
        elif elem.name == 'table':
            print("TABLE FOUND:")
            for row in elem.find_all('tr'):
                cols = [td.get_text(strip=True) for td in row.find_all(['td', 'th'])]
                print("  ROW:", cols)
        else:
            text = elem.get_text(strip=True)
            if text:
                print(f"{elem.name.upper()}: {text}")
else:
    print("card-body div not found!")

File: scratch/extract_clean_brochures.php
<?php
$html = file_get_contents('d:/xampp/htdocs/sssu/satya-sai/assets/images/sssutms.co.in/cms/Website/Admission/Brochures.html');

$dom = new DOMDocument();
@$dom->loadHTML($html);

$xpath = new DOMXPath($dom);
$cards = $xpath->query('//div[contains(@class, "card-body")]');

if ($cards->length > 0) {
    $card = $cards->item(0);
    
    // Remove base64 images so output stays clean
    $imgs = $card->getElementsByTagName('img');
    for ($i = $imgs->length - 1; $i >= 0; $i--) {
        $img = $imgs->item($i);
        $src = $img->getAttribute('src');
        if (strpos($src, 'data:image') === 0) {
            $img->setAttribute('src', '[BASE64_IMAGE]');
        }
    }
    
    // Print all text elements
    $nodes = $xpath->query('.//h1 | .//h2 | .//h3 | .//h4 | .//h5 | .//h6 | .//p | .//a | .//table', $card);
    foreach ($nodes as $node) {
        if ($node->nodeName == 'a') {
            $href = $node->getAttribute('href');
            $text = trim(preg_replace('/\s+/', ' ', $node->textContent));
            echo "LINK: Text='{$text}' | Href='{$href}'\n";
        } elseif ($node->nodeName == 'table') {
            echo "--- TABLE ---\n";
            $rows = $node->getElementsByTagName('tr');
            foreach ($rows as $row) {
                $cells = [];
                foreach ($row->childNodes as $cell) {
                    if ($cell->nodeName == 'td' || $cell->nodeName == 'th') {
                        $cells[] = trim(preg_replace('/\s+/', ' ', $cell->textContent));
                    }
                }
                echo "ROW: " . implode(" | ", $cells) . "\n";
            }
        } else {
            $text = trim(preg_replace('/\s+/', ' ', $node->textContent));
            if (!empty($text)) {
                echo strtoupper($node->nodeName) . ": {$text}\n";
            }
        }
    }
}
